tests/test_sql_query.py

```diff
@@ -871,6 +871,53 @@ async def test_sql_query_44(self):
         ):
             await self.squery_values('SELECT name FROM User')
 
+    async def test_sql_query_45(self):
+        with self.assertRaisesRegex(
+            asyncpg.InvalidColumnReferenceError,
+            'duplicate column name: `a`',
+            position="16",
+        ):
+            await self.squery_values('SELECT 1 AS a, 2 AS a')
+
+    async def test_sql_query_46(self):
+        res = await self.scon.fetch(
+            '''
+            WITH
+              x(a) AS (VALUES (1)),
+              y(a) AS (VALUES (2)),
+              z(a) AS (VALUES (3))
+            SELECT * FROM x, y JOIN z u on TRUE
+            '''
+        )
+
+        # `a` would be duplicated,
+        # so second and third instance are prefixed with rel var name
+        self.assert_shape(res, 1, ['a', 'ya', 'ua'])
```

Comment:
Do we want `xa`, `ya` and `ua` instead?

(it was a bit easier to do it this way)

Reply:
I think we want an underscore, so `y_a`, `u_a` etc.